### PR TITLE
Add creation time support to `FileTimes` on apple and windows

### PR DESCRIPTION
**Pull request by beetrees**
Adds support for setting file creation times on platforms which support changing it directly (currently only Apple and Windows). Based on top of #110093 (which was split from this PR).

ACP: rust-lang/libs-team#199 (currently still in progress)

Tracking issue: #98245

@rustbot label +T-libs-api -T-libs

### CODE REVIEW

**Comment by cuviper** [comment]
> Also refactors existing support for 64-bit `time_t` on 32-bit glibc platforms, as riscv32 `time_t` is 64-bit by default.

That part looks fine with me, and might deserve an independent PR. You should create an ACP for the API addition.
r? libs-api

File: library/std/src/sys/unix/fs.rs [hunk 1]
```diff
@@ -1231,6 +1252,22 @@ impl File {
                 })?;
                 Ok(())
             } else {
+                #[cfg(all(target_os = "linux", target_env = "gnu", target_pointer_width = "32", not(target_arch = "riscv32")))]
```
Comment (Amanieu) [inline library/std/src/sys/unix/fs.rs:1255]: These changes are to 32-bit RISC-V Linux and seem to be unrelated to this PR?

Comment (Amanieu) [reply library/std/src/sys/unix/fs.rs:1255]: Yes, this should be split into a separate PR since it doesn't involve any API changes.

**Comment by Amanieu** [comment]
Thanks! You still need to remove the RISC-V commit from this PR.

Could you add a test in library/std/src/fs/tests.rs? Just try setting the file time and reading it back to ensure it worked.

**Comment by Amanieu** [comment]
LGTM! This PR is still marked as a draft, are there any more changes or is it ready to merge?

**Comment by ehuss** [comment]
@bors retry

crates.io network blip

**Comment by ehuss** [comment]
@bors retry

crates.io network blip

**Comment by lqd** [comment]
I believe this to be noise that has since then returned to steady state. This PR being the rising edge of the last bump in the following graphs.

image

image


@rustbot label: +perf-regression-triaged
